amcl:
  ros__parameters:
    use_sim_time: false
    alpha1: 0.2
    alpha2: 0.2
    alpha3: 0.2
    alpha4: 0.2
    alpha5: 0.2
    base_frame_id: base_link
    odom_frame_id: odom
    global_frame_id: map
    scan_topic: /scan
    min_particles: 500
    max_particles: 2000
    kld_err: 0.05
    kld_z: 0.99
    update_min_d: 0.25
    update_min_a: 0.2
    resample_interval: 1
    transform_tolerance: 0.5
    recovery_alpha_slow: 0.0
    recovery_alpha_fast: 0.0
    gui_publish_rate: 10.0
    max_beams: 60
    max_range: 8.0
    z_hit: 0.5
    z_short: 0.05
    z_max: 0.05
    z_rand: 0.5
    sigma_hit: 0.2
    lambda_short: 0.1
    laser_model_type: likelihood_field
    odom_model_type: diff
    init_pose_x: 0.0
    init_pose_y: 0.0
    init_pose_a: 0.0
    init_cov_xx: 0.25
    init_cov_yy: 0.25
    init_cov_aa: 0.0685

bt_navigator:
  ros__parameters:
    use_sim_time: false
    default_nav_to_pose_bt_xml: "navigate_w_replanning_and_recovery.xml"

controller_server:
  ros__parameters:
    use_sim_time: false
    controller_plugins: ["FollowPath"]
    FollowPath:
      plugin: "dwb_core::DWBLocalPlanner"
      debug_trajectory_details: false
    min_vel_x: 0.0
    max_vel_x: 0.26
    max_vel_theta: 1.0
    min_speed_xy: 0.0
    max_speed_xy: 0.26
    min_speed_theta: -1.0
    acc_lim_x: 2.5
    acc_lim_theta: 3.2
    decel_lim_x: -2.5
    decel_lim_theta: -3.2
    vx_samples: 20
    vtheta_samples: 40
    sim_time: 1.7
    linear_granularity: 0.05
    angular_granularity: 0.025
    goal_distance_bias: 24.0
    path_distance_bias: 32.0
    occdist_scale: 0.01
    stop_time_buffer: 0.2
    forward_point_distance: 0.325
    scaling_speed: 0.25
    max_scaling_factor: 0.2

planner_server:
  ros__parameters:
    use_sim_time: false
    expected_planner_frequency: 1.0
    planner_plugins: ["GridBased"]
    GridBased:
      plugin: "nav2_navfn_planner/NavfnPlanner"

map_server:
  ros__parameters:
    use_sim_time: false
    yaml_filename: ""

behavior_server:
  ros__parameters:
    use_sim_time: false
    behavior_plugins: ["spin", "backup", "wait"]
    spin:
      plugin: "nav2_behaviors/Spin"
    backup:
      plugin: "nav2_behaviors/BackUp"
    wait:
      plugin: "nav2_behaviors/Wait"

local_costmap:
  ros__parameters:
    use_sim_time: false
    global_frame: odom
    robot_base_frame: base_link
    update_frequency: 5.0
    publish_frequency: 2.0
    rolling_window: true
    width: 3.0
    height: 3.0
    resolution: 0.05
    plugins: ["voxel_layer", "inflation_layer"]
    voxel_layer:
      plugin: "nav2_costmap_2d::VoxelLayer"
    inflation_layer:
      plugin: "nav2_costmap_2d::InflationLayer"

global_costmap:
  ros__parameters:
    use_sim_time: false
    global_frame: map
    robot_base_frame: base_link
    update_frequency: 5.0
    publish_frequency: 2.0
    width: 20.0
    height: 20.0
    resolution: 0.05
    plugins: ["static_layer", "obstacle_layer", "inflation_layer"]
    static_layer:
      plugin: "nav2_costmap_2d::StaticLayer"
    obstacle_layer:
      plugin: "nav2_costmap_2d::ObstacleLayer"
    inflation_layer:
      plugin: "nav2_costmap_2d::InflationLayer"

recoveries_server:
  ros__parameters:
    use_sim_time: false
    recovery_plugins: ["spin", "backup", "wait"]
    spin:
      plugin: "nav2_behaviors/Spin"
    backup:
      plugin: "nav2_behaviors/BackUp"
    wait:
      plugin: "nav2_behaviors/Wait"
